### PR TITLE
Update to use WasmEdge-0.9.0-rc.2 core.

### PR DESCRIPTION
**Pull request by q82419**

### CODE REVIEW

File: wasmedge/executor.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		return newError(res)
+	}
+	return nil
+}
+
+func (self *Executor) Invoke(store *Store, funcname string, params ...interface{}) ([]interface{}, error) {
+	funcstr := toWasmEdgeStringWrap(funcname)
+	funccxt := store.FindFunction(funcname)
+	ftype := funccxt.GetFunctionType()
+	cparams := toWasmEdgeValueSlide(params...)
```
Comment (hydai) [inline wasmedge/executor.go:84]: Does `cparams` need to be deleted at the end of this function?

**Review by hydai — CHANGES_REQUESTED** [review]

File: wasmedge/vm.go [hunk 2]
```diff
@@ -296,36 +319,32 @@ func (self *VM) GetFunctionList() ([]string, []*FunctionType) {
 	ftypes := make([]*FunctionType, int(funclen))
 	for i := 0; i < int(funclen); i++ {
 		fnames[i] = fromWasmEdgeString(cfnames[i])
-		C.WasmEdge_StringDelete(cfnames[i])
```
[the remark this answers is not in the document]
Comment (q82419) [reply wasmedge/vm.go:299]: In the newer `0.9.0-rc.2` core, the deletion is not needed.

File: wasmedge/store.go [hunk 3]
```diff
@@ -199,12 +200,15 @@ func (self *Store) ListModule() []string {
 	names := make([]string, int(modlen))
 	for i := 0; i < int(modlen); i++ {
 		names[i] = fromWasmEdgeString(cnames[i])
-		C.WasmEdge_StringDelete(cnames[i])
```
[the remark this answers is not in the document]
Comment (q82419) [reply wasmedge/store.go:202]: In the newer `0.9.0-rc.2` core, the deletion is not needed.

File: wasmedge/executor.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+	runtime.SetFinalizer(res, (*Executor).Release)
+	return res
+}
+
+func (self *Executor) Instantiate(store *Store, ast *AST) error {
+	res := C.WasmEdge_ExecutorInstantiate(self._inner, store._inner, ast._inner)
+	if !C.WasmEdge_ResultOK(res) {
+		return newError(res)
+	}
+	return nil
```
[the remark this answers is not in the document]
Comment (q82419) [reply wasmedge/executor.go:60]: We return `nil` for the `WasmEdge_Result` is no error in all functions.

File: wasmedge/executor.go (hunk 1 above)
Comment (q82419) [reply wasmedge/executor.go:84]: `WasmEdge_Value` not needs to be deleted.

File: wasmedge/store.go [hunk 5]
```diff
@@ -39,19 +41,18 @@ func (self *Store) getExports(exportlen C.uint32_t, getfunc interface{}, modname
 	names := make([]string, int(exportlen))
 	for i := 0; i < int(exportlen); i++ {
 		names[i] = fromWasmEdgeString(cnames[i])
-		C.WasmEdge_StringDelete(cnames[i])
```
[the remark this answers is not in the document]
Comment (q82419) [reply wasmedge/store.go:42]: The `WasmEdge_String` gets from `Store`, `VM`, or other objects is a wrap to string and cannot be deleted.

File: .CurrentChangelog.md [hunk 6]
```diff
@@ -1,5 +1,71 @@
...
+    * Renamed `(*Configure).IsCompilerCostMeasuring` to `(*Configure).IsStatisticsCostMeasuring`.
+  * Statistics
+    * Renamed `(*Statistics).Delete` to `(*Statistics).Release`.
+  * Compiler
+    * Renamed `(*Compiler).Delete` to `(*Compiler).Release`.
+  * Loader
+    * Renamed `(*Loader).Delete` to `(*Loader).Release`.
+  * Validator
+    * Renamed `(*Validator).Delete` to `(*Validator).Release`.
+  * Interpreter: Renamed to `Executor`
```
[the remark this answers is not in the document]
Comment (q82419) [reply .CurrentChangelog.md]: Fixed.

**Review by hydai — APPROVED** [review]

**Pull request merged**